File: 0274-h-index/0274-h-index.c
int cmpfunc(const void *a, const void *b)
{
    return(*(int*)a - *(int*)b);
}

int hIndex(int* citations, int citationsSize){

    qsort(citations, citationsSize, sizeof(int), cmpfunc);

    if (citations[0] >= citationsSize){
        return citationsSize;
    } else if (citationsSize > 1){
        return hIndex(&citations[1], citationsSize-1);
    }

    return 0;
}

File: 0274-h-index/0274-h-index.py
class Solution(object):
    def hIndex(self, citations):
        """
        :type citations: List[int]
        :rtype: int
        """
        counts = [0 for _ in range(1001)]
        for p in citations:
            counts[p] += 1

        total_papers = 0
        for i in range(1000, -1, -1):
            total_papers += counts[i]
            if total_papers >= i:
                return i
        
        return 0
        